src/check_cache.cc

```diff
@@ -51,8 +51,7 @@ bool CheckCache::ShouldFlush(const CacheElem& elem) {
   return age >= flush_interval_in_cycle_;
 }
 
-Status CheckCache::Check(const Attributes& attributes, CheckResponse* response,
-                         std::string* signature) {
+Status CheckCache::Check(const Attributes& attributes, std::string* signature) {
```

Comment:
I don't see where the code is checking for the expiration count in the cache.

A given cached response should only be used for the specified amount of time, and for the specified number of uses.